lib/backend/pgbk/pgbk.go

```diff
@@ -159,16 +154,18 @@ func NewWithConfig(ctx context.Context, cfg Config) (*Backend, error) {
 	log := logrus.WithField(trace.Component, componentName)
 
 	if cfg.AuthMode == AzureADAuth {
-		bc, err := pgcommon.AzureBeforeConnect(cfg.AzureClientID, log)
+		bc, err := pgcommon.AzureBeforeConnect(log)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
 		poolConfig.BeforeConnect = bc
 	}
 
-	poolConfig.AfterConnect = func(ctx context.Context, c *pgx.Conn) error {
-		_, err := c.Exec(ctx, "SET default_transaction_isolation TO serializable", pgx.QueryExecModeExec)
-		return trace.Wrap(err)
+	if poolConfig.ConnConfig.RuntimeParams["default_transaction_isolation"] != "" {
+		log.WithField("default_transaction_isolation", poolConfig.ConnConfig.RuntimeParams["default_transaction_isolation"]).
+			Warn("The default_transaction_isolation parameter was overridden in the connection string; proceeding with an unsupported configuration.")
+	} else {
+		poolConfig.ConnConfig.RuntimeParams["default_transaction_isolation"] = "serializable"
```

Comment:
nit: 
```suggestion
		poolConfig.ConnConfig.RuntimeParams["default_transaction_isolation"] = pgx.Serializable
```

Reply:
I disagree with that, `pgx.Serializable` only happens to be the `"serializable"` string (behind a newtype, so it would have to be `string(pgx.Serializable)` in our case) for internal `pgx` convenience - it's used for string formatting/concatenation in places, internally - but it wouldn't be a compatibility break for `pgx` to change it, as it's only intended to be passed in `pgx.TxOptions.IsoLevel`.